sentry/src/main/java/io/sentry/Hub.java

```diff
@@ -699,21 +699,12 @@ public void flush(long timeoutMillis) {
     boolean samplingDecision = tracingSampler.sample(samplingContext);
     transactionContexts.setSampled(samplingDecision);
 
-    SentryTransaction transaction = null;
-    if (!isEnabled()) {
-      options
-          .getLogger()
-          .log(
-              SentryLevel.WARNING,
-              "Instance is disabled and this 'startTransaction' call is a no-op.");
+    SentryTransaction transaction = new SentryTransaction(transactionContexts, this);
```

Comment:
Can we return `NoOpTransaction` if SDK is disabled instead?

If the goal is to have no overhead, we can avoid the allocations.